src/transforms/filter.js

```diff
@@ -190,85 +187,71 @@ function getFilterFunc(opts, d2c) {
-                    return cv > coercedValue[0] && cv < coercedValue[1];
-                };
-            }
-            else if(coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv > coercedValue[0] && cv <= coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv >= coercedValue[0] && cv < coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv >= coercedValue[0] && cv <= coercedValue[1];
-                };
-            }
-
-            break;
-
-        case 'notwithin':
-
-            if(coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv < coercedValue[0] || cv > coercedValue[1];
-                };
-            }
-            else if(coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv < coercedValue[0] || cv >= coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv <= coercedValue[0] || cv > coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv <= coercedValue[0] || cv >= coercedValue[1];
-                };
-            }
-
-            break;
-
-        case 'in':
+            return function(v) { return d2c(v) > coercedValue; };
+
+        case '>=':
+            return function(v) { return d2c(v) >= coercedValue; };
+
+        case '[]':
+            return function(v) {
+                var cv = d2c(v);
+                return cv >= coercedValue[0] && cv <= coercedValue[1];
+            };
+
+        case '()':
+            return function(v) {
+                var cv = d2c(v);
+                return cv > coercedValue[0] && cv < coercedValue[1];
+            };
+
+        case '[)':
+            return function(v) {
+                var cv = d2c(v);
+                return cv >= coercedValue[0] && cv < coercedValue[1];
+            };
+
+        case '(]':
+            return function(v) {
+                var cv = d2c(v);
+                return cv > coercedValue[0] && cv <= coercedValue[1];
+            };
+
+        case '][':
+            return function(v) {
+                var cv = d2c(v);
+                return cv < coercedValue[0] || cv > coercedValue[1];
```

Comment:
btw `][` means 'not in the _open_ interval and code is correctly implemented that way